File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) { int res=0; int cnt=0; 
        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='(')
            {
                cnt++;
                res=max(cnt,res);
            }
            if(s[i]==')') cnt--; 
        }
        return res;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) { int cnt=0; int ans=0;
        for(int i=0;i<s.length();i++)
        {
            char ch=s.charAt(i);
            if(ch=='(') cnt++;
            else if(ch==')') cnt--;
            ans=Math.max(ans,cnt);
        }
        return ans; 
    }
}